### ISSUE TITLE
Idea: use the RGB8 type from the rgb crate

### ISSUE THREAD

**Issue by silwol** [smart-leds-rs/smart-leds#4]
The `rgb` crate provides a `RGB8` type alias which corresponds to this crate's `Color` struct. Given the fact that some programs might have the `RGB8` in use already, I think it makes sense to use it here as well.

**Comment by david-sawatzke** [comment]
The type alias should be in [smart-leds-trait](https://github.com/smart-leds-rs/smart-leds-trait).

Not sure about adding a dependency to it (although `rgb` does have delightfully little additional dependencies. Namely none). 

Additionally, we might also add a type for rgbw leds in the future and I'm not sure if it's a good idea to have two wholly different types, that mostly don't share code. 

But the RGB8 crate does have quite a few nice convenience functions that would support our "don't give us a buffer, just do iterators" api & interop is definitely nicer, although it doesn't seem to be used much in embedded stuff currently.

I'm tending towards including it. @sajattack, what do you think?

**Comment by sajattack** [comment]
Yeah, I like it. Isn't RGBW the same as RGBA?

**Comment by david-sawatzke** [comment]
Nah, rgba is the alpha value, more like a brightness in our context I think. rgbw has a separate white value that "should" be contained in the rgb value.

Probably not a bad idea to put it in rgb, don't know if it's going to be accepted.

**Comment by sajattack** [comment]
I've filed an issue with them to see what they think https://github.com/kornelski/rust-rgb/issues/21

**Comment by sajattack** [comment]
Could we type alias rgba to rgbw?

**Comment by david-sawatzke** [comment]
kornelski/rust-rgb#21 The rgbw type is needed because there exist leds that have a separate white led. So you have two ways to archieve white, that theoretically the same color. E.g. w: 0xFF ~ r:0xFF, g:0xFF, b:0xFF, so that type's not really a great fit for the rgb crate.

The best thing would probably to just copy the rgba stuff and do a regex type replace from rgba to rgbw, since they do often (but not always) behave the same.
